### ISSUE TITLE
Add Carla search location to LMMS settings in vindows

### ISSUE THREAD

**Issue by Rossmaxx** [LMMS/lmms#6802]
### Enhancement Summary
It would be a good idea to search for carla in a user defined settings (either via gui or the `lmmsrc.xml` file) in windows.

#### Justification
Currently, LMMS uses the `PATH` environment variable to search for carla. Using PATH is unintuitive for new users looking to use carla with lmms on windows and also got some minor technical issues.

I'm talking about adding carla path as an entry here.
![image](https://github.com/LMMS/lmms/assets/74815851/ef5059b2-46d4-4954-9a1c-ee604afe4469)

**Comment by Rossmaxx** [comment]
pinging @falkTX and @tresf for discussion.

**Comment by tresf** [comment]
> pinging falkTX and @tresf for discussion.

This is more of an LMMS task, so no need to ping the Carla author I don't think. 😅

**Comment by Rossmaxx** [comment]
> This is more of an LMMS task, so no need to ping the Carla author I don't think. 😅

Oops mb.

> Currently on Mac, a similar problem exists.

I thought the issue was specific to windows

> I would ask that it's extended to work for other OSs as well.

So this would be added to mac as well. Am i correct?

**Comment by tresf** [comment]
> > This is more of an LMMS task, so no need to ping the Carla author I don't think. 😅
> 
> Oops mb.
> 
> > Currently on Mac, a similar problem exists.
> 
> I thought the issue was specific to windows
> 
> > I would ask that it's extended to work for other OSs as well.
> 
> So this would be added to mac as well. Am i correct?

Yes.  Specifically, if we can find a way to get it working for All OSs, that would be best.  Currently Carla supports the following OSs:

* Linux 32bit
* Linux 64bit
* MacOS universal build (requires macOS 10.12 or higher)
* Windows 32bit
* Windows 64bit

... So I would say if `WANT_CARLA` is enabled, this feature should be on.  Our existing logic of searching for Carla in `../Applications` and `$PATH` should be safe to keep as a fallback.

**Comment by falkTX** [comment]
Carla can work on any POSIX based system, that includes BSD, HaikuOS, GNU Hurd and even web assembly via emscripten. The GUI needs PyQt for now, but backend is quite flexible
